packages/web-components/src/components/rux-modal/rux-modal.tsx

```diff
@@ -50,36 +61,52 @@ export class RuxModal {
     })
     ruxModalClosed!: EventEmitter<boolean>
 
+    /**
+     * Event that is fired when modal opens
+     */
+    @Event({
+        eventName: 'ruxmodalopened',
+        composed: true,
+        bubbles: true,
+    })
+    ruxModalOpened!: EventEmitter<boolean>
+
     @Element() element!: HTMLRuxModalElement
 
-    // confirm dialog if Enter key is pressed
-    @Listen('keydown', { target: 'window' })
-    handleKeyDown(ev: KeyboardEvent) {
-        if (ev.key === 'Enter') {
-            const button = this._getDefaultButton()
-            if (button) {
-                button.click()
-            }
+    @Watch('open')
+    async handleOpenChange() {
+        if (this.open) {
+            this.ruxModalOpened.emit(true)
+        } else {
+            this.ruxModalClosed.emit(true)
         }
     }
 
     // close modal if click happens outside of dialog
     @Listen('click', { target: 'window' })
     handleClick(ev: MouseEvent) {
-        const wrapper = this._getWrapper()
-        if (ev.composedPath()[0] === wrapper) {
-            this.ruxModalClosed.emit(false)
-            this.open = false
+        if (this.clickToClose) {
+            const wrapper = this._getWrapper()
+            if (ev.composedPath()[0] === wrapper) {
+                this.ruxModalClosed.emit(false)
+                this.open = false
+            }
         }
     }
 
-    @Watch('open')
-    validateName(isOpen: boolean) {
-        if (isOpen) {
+    connectedCallback() {
+        this._handleModalChoice = this._handleModalChoice.bind(this)
+    }
+
+    componentDidRender() {
```

Comment:
This lifecycle method handles the auto-focusing much better. We wouldn't even need setTimeout in the _focusDefaultButton method, but it still needs it when modal is set to open by default. Otherwise, if modal is being opened by some action, the focus worked without the setTimeout.